operator/apis/platform/v1alpha1/securedcluster_types.go

```diff
@@ -291,7 +291,7 @@ type CollectorContainerSpec struct {
 	//+operator-sdk:csv:customresourcedefinitions:type=spec,order=2
 	ImageFlavor *CollectorImageFlavor `json:"imageFlavor,omitempty"`
 
-	// Set this to 'true' to prevent translation of the collection method, e.g. to enforce using EBPF as an exception.
+	// Set this to 'true' to prevent conversion of the collection method, e.g. to enforce using EBPF as an exception.
```

Comment:
Can you please show how this looks like in the openshift UI? TBH as a user I would have no idea what this line means. I don't even know what it means as an ACS developer :sweat_smile:

Reply:
It would take me some time to get a screenshot. In the meantime, do you have any suggestions how to formulate it better?

Reply:
I honestly don't know what this achieves, let me take a look at the code...

Reply:
@porridge some context.

The CollectionMethod can have several values:
* CORE_BPF (default)
* NoCollection
* EBPF (deprecated)
* KernelModules (removed)

EBPF and KernelModules are converted by the operator to `CORE_BPF`. This flag would force `EBPF` to not be converted, for customers that cannot use `CORE_BPF`. To add to the confusion, it wouldn't work on `KernelModules` because that method is not available at all anymore.

Reply:
If [this](https://github.com/stackrox/stackrox/blob/8ddb13cd0d90fe6756419afde12cb163aeaf5e0b/operator/pkg/securedcluster/values/translation/translation.go#L286C1-L292) is all that this controls, then I'd probably say something like:
```suggestion
	// When the above 'collection' field is set to the deprecated value 'EBPF', then setting this field to 'true' prevents treating it as if it was set to 'CORE_BFP'.
```

But I'd probably add another sentence describing when this is appropriate.

Reply:
> treating it as if it was set to 'CORE_BFP'.

This sounds very generic, and most likely triggers questions like "threating in which sense" etc. I would still use "conversion" here.

> add another sentence describing when this is appropriate.

This part of the original sentence was doing exactly that: "to enforce using EBPF as an exception".

Reply:
I think "conversion" would make sense, as long that word is also used in the description of the collection field, and we keep pointing in that direction? :thinking:

> > add another sentence describing when this is appropriate.
> This part of the original sentence was doing exactly that: "to enforce using EBPF as an exception".

I meant to say that this seems very terse.  When do you think such exception would make sense for example?

Reply:
> I think "conversion" would make sense, as long that word is also used in the description of the collection field, and we keep pointing in that direction? 🤔

Yep, I can elaborate about conversion in the "collection" field description.

>  When do you think such exception would make sense for example?

Troubleshooting of the collection method, or workaround if it doesn't work under certain circumstances -- this is essentially an internal fallback mechanism just in case. But I think these details go beyond a reasonable description, especially taking into account it's mostly for us internally, and most users would not and should not modify this value.

Reply:
Perhaps we should hide that field with `xDescriptors={"urn:alm:descriptor:com.tectonic.ui:hidden"}`, then like we do for a couple of fields elsewhere in this file?

Reply:
This would hide it from the UI, right? Didn't know something like this is possible, sounds good to me.